batchflow/models/torch/base.py

```diff
@@ -1555,112 +1542,103 @@ def aggregate_microbatches(self, outputs, chunked_outputs, chunk_sizes, single_o
+            else:
+                result[name] = self.apply_output_operation(predictions, operation)
+        return result
 
     @staticmethod
-    def apply_output_operation(tensor, operation):
-        """ Apply `operation`, possibly aliased with a string, to `tensor`. """
+    def apply_output_operation(predictions, operation):
+        """ Apply `operation`, possibly aliased with a string, to `predictions`. """
         with torch.no_grad():
             if operation is None:
-                result = tensor
-                name = ''
+                result = predictions
             elif callable(operation):
-                result = operation(tensor)
-                name = operation.__name__
+                result = operation(predictions)
+            elif isinstance(operation, LayerHook):
+                operation.close()
+                result = operation.activation
+            elif (isinstance(operation, str) and # Check if operation matches `predictions_[0-9]+`
+                  operation.startswith('predictions_') and
+                  operation.split("_")[-1].isdigit()):
+
+                i = int(operation.split("_")[-1])
+                result = predictions[i]
             else:
+                if isinstance(predictions, (tuple, list)) and not len(predictions) == 1:
+                    raise ValueError('Default operations can`t be applied to multi output predictions.')
+
+                predictions = predictions[0] if isinstance(predictions, (tuple, list)) else predictions
                 if operation == 'softplus':
-                    result = torch.nn.functional.softplus(tensor) # pylint: disable=not-callable
+                    result = torch.nn.functional.softplus(predictions) # pylint: disable=not-callable
                 elif operation == 'sigmoid':
-                    result = torch.sigmoid(tensor)
+                    result = torch.sigmoid(predictions)
                 elif operation == 'sigmoid_uint8':
-                    result = (torch.sigmoid(tensor) * 255).to(dtype=torch.uint8)
+                    result = (torch.sigmoid(predictions) * 255).to(dtype=torch.uint8)
                 elif operation == 'sigmoid_int16':
-                    result = (torch.sigmoid(tensor) * 255).to(dtype=torch.int16)
+                    result = (torch.sigmoid(predictions) * 255).to(dtype=torch.int16)
                 elif operation == 'proba':
-                    result = torch.nn.functional.softmax(tensor, dim=1)
+                    result = torch.nn.functional.softmax(predictions, dim=1)
                 elif operation == 'labels':
-                    result = tensor.argmax(dim=1)
+                    result = predictions.argmax(dim=1)
                 else:
                     raise ValueError(f'Unknown type of operation `{operation}`!')
-                name = operation
-        return result, name
+        return result
 
 
     def prepare_outputs(self, outputs):
```

Comment:
`prepare_outputs` sounds very much alike to `compute_outputs`, while the intent here is to `prepare user-passed argument "outputs" to a form that we internally use". Do you see any better names?